mesonbuild/build.py

```diff
@@ -1113,9 +1119,49 @@ def __init__(self, name, subdir, subproject, is_cross, sources, objects, environ
             self.filename += '.' + self.suffix
         self.outputs = [self.filename]
 
+        # The import library this target will generate
+        self.import_filename = None
+        # The import library that Visual Studio would generate (and accept)
+        self.vs_import_filename = None
+        # The import library that GCC would generate (and prefer)
+        self.gcc_import_filename = None
+
+        # if implib:true appears, this target is linkwith:-able, but that only
+        # means something on Windows platforms.
+        self.is_linkwithable = False
+        if 'implib' in kwargs and kwargs['implib']:
+            self.is_linkwithable = True
+            if for_windows(is_cross, environment) or for_cygwin(is_cross, environment):
+                self.vs_import_filename = '{0}.lib'.format(self.name)
```

Comment:
Yeah.  

Note that the implib isn't going to be generated unless explicitly requested, so it is still ok to have a .dll and .exe with the same basename, provided implib: isn't used.

I'm unsure of the conventions for naming this import library with MSVC, but I believe usages like this are seen in the wild.

I guess control over the implib name is needed, with a default which avoids collisions.